src/Microsoft.Diagnostics.EventFlow.Inputs.Etw/StandardTraceEventSession.cs

```diff
@@ -74,17 +75,23 @@ public void Process(Action<EventData> onEvent)
             if (!isProcessing)
             {
                 isProcessing = true;
-                this.inner.Source.Dynamic.All += (traceEvent) => 
-                {
-                    // Suppress events from TplEventSource--they are mostly interesting for debugging task processing and interaction,
-                    // and not that useful for production tracing. However, TPL EventSource must be enabled to get hierarchical activity IDs.
-                    if (!TplActivities.TplEventSourceGuid.Equals(traceEvent.ProviderGuid))
-                    {
-                        onEvent(traceEvent.ToEventData(this.healthReporter));
-                    }
-                };
+                this.onEvent = onEvent;
+
+                this.inner.Source.Dynamic.All += TrackEvent;
```

Comment:
Is it the case that the event handler will be accumulated if Process() is called multiple times? Looks like a potential memory leaking spot. Maybe remove the handler before adding it back?

Reply:
@xiaomi7732 that is not possible--`isProcessing` member variable is there to prevent that